### PR TITLE
Restrict admins from self-removal

### PR DESCRIPTION
**Pull request by rhafer**
Admin users are no longer allowed to remove their own account or
to edit their own role assigments. By this restriction we try to
prevent situation where no administrative users is available
in the system anymore

Fixes: #3713

### CODE REVIEW

File: services/settings/pkg/service/v0/service.go [hunk 1]
```diff
@@ -385,6 +385,15 @@ func (g Service) AssignRoleToUser(ctx context.Context, req *settingssvc.AssignRo
 	if validationError := validateAssignRoleToUser(req); validationError != nil {
 		return merrors.BadRequest(g.id, "%s", validationError)
 	}
+
+	ownAccountUUID, ok := metadata.Get(ctx, middleware.AccountID)
+	if !ok {
+		return merrors.InternalServerError(g.id, "user not in context")
+	}
+	if ownAccountUUID == req.AccountUuid {
+		return merrors.Forbidden(g.id, "%s", "Removing own role assignment forbidden")
```
Comment (micbar) [inline services/settings/pkg/service/v0/service.go:394]: Wrong error message? Maybe `Changing own role assignment forbidden`?

Comment (micbar) [reply services/settings/pkg/service/v0/service.go:394]: And what about logging this on debug level?

File: services/settings/pkg/service/v0/service.go [hunk 2]
```diff
@@ -399,9 +408,26 @@ func (g Service) RemoveRoleFromUser(ctx context.Context, req *settingssvc.Remove
 		return err
 	}
 
+	ownAccountUUID, ok := metadata.Get(ctx, middleware.AccountID)
+	if !ok {
+		return merrors.InternalServerError(g.id, "user not in context")
```
Comment (micbar) [inline services/settings/pkg/service/v0/service.go:413]: log also?

File: services/settings/pkg/service/v0/service.go [hunk 3]
```diff
@@ -399,9 +408,26 @@ func (g Service) RemoveRoleFromUser(ctx context.Context, req *settingssvc.Remove
...
+	}
+
+	al, err := g.manager.ListRoleAssignments(ownAccountUUID)
+	if err != nil {
+		return merrors.InternalServerError(g.id, "%s", err)
```
Comment (micbar) [inline services/settings/pkg/service/v0/service.go:418]: log?

File: services/settings/pkg/service/v0/service.go [hunk 4]
```diff
@@ -399,9 +408,26 @@ func (g Service) RemoveRoleFromUser(ctx context.Context, req *settingssvc.Remove
...
+	}
+
+	for _, a := range al {
+		if a.Id == req.Id {
+			return merrors.Forbidden(g.id, "%s", "Removing own role assignment forbidden")
```
Comment (micbar) [inline services/settings/pkg/service/v0/service.go:423]: log?

**Review by micbar — COMMENTED** [review]
Some nitpicks 😄

**Review by micbar — APPROVED** [review]

**Review by C0rby — APPROVED** [review]

**Pull request merged**